honestcash-v2/src/app/wallet/services/wallet.service.ts

```diff
@@ -21,65 +21,108 @@ export const WALLET_LOCALSTORAGE_KEYS = {
 export class WalletService {
-  public _wallet: Wallet = null;
-  public wallet: Wallet = null;
-
+  public wallet: ISimpleWallet;
   public isSettingUpWallet: Subject<WALLET_SETUP_STATUS> = new AsyncSubject<WALLET_SETUP_STATUS>();
-
   private logger: Logger;
-  readonly isPlatformBrowser: boolean;
+  private readonly isPlatformBrowser: boolean;
 
   constructor(
     @Inject(PLATFORM_ID) private platformId: any,
     @Inject(LocalStorageToken) private localStorage: Storage,
     private http: HttpService,
-    private authenticationService: AuthService,
+    private scriptService: ScriptService,
   ) {
     this.logger = new Logger('WalletService');
     this.isPlatformBrowser = isPlatformBrowser(this.platformId);
     this.isSettingUpWallet.next(WALLET_SETUP_STATUS.NotInitialized);
   }
 
-  public setupWallet(payload?: LoginSuccessResponse | SignupSuccessResponse): Observable<ISimpleBitcoinWallet | Error> {
+  public static calculateSHA3Hash(message: string): string {
+    return sha3_512(message);
+  }
+
+  public static determineMessageForHashing(salt: string, password: string): string {
+    return `${salt}:${password}`;
+  }
+
+  public static calculatePasswordHash(email: string, password: string): string {
+    return WalletService.calculateSHA3Hash(
+      WalletService.determineMessageForHashing(email, password)
+    );
+  }
+
+  public createWallet(password: string): ISimpleWallet {
+    return new SimpleWallet(null, {password});
+  }
+
+  public encrypt(mnemonic: string, password: string): string {
+    return SimpleWallet.encrypt(mnemonic, password);
+  }
+
+  public decrypt(mnemonicEncrypted: string, password: string): string {
+    return SimpleWallet.decrypt(mnemonicEncrypted, password);
+  }
+
+  public loadWalletWithEncryptedRecoveryPhrase(
+    encryptedRecoveryPhrase: string,
+    password: string
+  ): ISimpleWallet {
+    return new SimpleWallet(encryptedRecoveryPhrase, {password});
+  }
+
+  public loadWalletWithDecryptedRecoveryPhrase(
+    recoveryPhrase: string,
+  ): ISimpleWallet {
+    return new SimpleWallet(recoveryPhrase, {password: null});
+  }
+
+  public loadWallet(payload?: LoginSuccessResponse): Observable<ISimpleWallet | WALLET_SETUP_STATUS> {
     return defer(
       async () => {
         this.isSettingUpWallet.next(WALLET_SETUP_STATUS.Started);
-        let simpleWallet: ISimpleBitcoinWallet;
+        await this.scriptService.loadScript(simpleBitcoinWalletAssetPath).toPromise();
+        let simpleWallet: ISimpleWallet;
         if (payload) {
           if ((<LoginSuccessResponse>payload).wallet && (<LoginSuccessResponse>payload).wallet.mnemonicEncrypted) {
             // if there is a payload and a wallet attached
             // it means it is a login action
             this.logger.info('Setting up an already existing wallet from payload');
-            simpleWallet = await WalletUtils.generateWalletWithEncryptedRecoveryPhrase(
+            simpleWallet = this.loadWalletWithEncryptedRecoveryPhrase(
```

Comment:
it was overkill to abstract these to a helper file as they are part of walletService. they can probably be private (not that it matters) but I'll still check if it is good to refactor these somewhere else after I understand that they work